pulp_container/app/registry_api.py

```diff
@@ -1300,6 +1300,8 @@ def put(self, request, path, pk=None):
             config_blob = found_config_blobs.first()
             manifest = self._init_manifest(manifest_digest, media_type, raw_text_data, config_blob)
             manifest.init_metadata(manifest_data=content_data)
+            manifest.init_architecture_and_os()
+            manifest.init_compressed_image_size()
```

Comment:
Do we now always need to call them adjacent to init_metadata?
Would it be useful to combine them into one to prevent holding it wrong?

Reply:
hmm... during the build and mirroring tasks these fields are not handled together because we don't have all the information to set them at the same time (they are gathered from different resources). So I don't know if merging them would be straightforward.

Reply:
Yeah, let's call it food for thought.

Reply:
We discussed this and agreed on opening a task issue to investigate this later.